modules/core/message_list_functions.php

```diff
@@ -351,13 +351,29 @@ function icon_callback($vals, $style, $output_mod) {
 if (!hm_exists('message_controls')) {
 function message_controls($output_mod) {
     $res = '<a class="toggle_link" href="#"><img alt="x" src="'.Hm_Image_Sources::$check.'" width="8" height="8" /></a>'.
-        '<div class="msg_controls">'.
-        '<a class="msg_read core_msg_control" href="#" data-action="read">'.$output_mod->trans('Read').'</a>'.
-        '<a class="msg_unread core_msg_control" href="#" data-action="unread">'.$output_mod->trans('Unread').'</a>'.
-        '<a class="msg_flag core_msg_control" href="#" data-action="flag">'.$output_mod->trans('Flag').'</a>'.
-        '<a class="msg_unflag core_msg_control" href="#" data-action="unflag">'.$output_mod->trans('Unflag').'</a>'.
-        '<a class="msg_delete core_msg_control" href="#" data-action="delete">'.$output_mod->trans('Delete').'</a>'.
-        '<a class="msg_archive core_msg_control" href="#" data-action="archive">'.$output_mod->trans('Archive').'</a>';
+        '<div class="msg_controls">
+            <div class="dynamic_dropdown on_mobile">
+              <button class="dynamic_dropdown-toggle" aria-haspopup="true" aria-expanded="false">
+                Actions
+                <span class="arrow"></span>
+              </button>
+              <ul class="dynamic_dropdown-menu" aria-hidden="true" aria-labelledby="dropdownMenuButton">
```

Comment:
Please take a look at class .cypht-dropdown and modify it's js so that we we have just one code of displaying modals.

Reply:
@josaphatim, I just saw .cypht-dropdown, I realize that it is used in one place and seems to be incomplete compared to what I have set up. different dropdown are used in different places (ex: filter, etc??).

What I suggest to you is that I create a task to standardize this. I think you will know that I left a TODO in the code to signify this.

Reply:
Just a note that cypht-dropdown was introduced in order to not interfere with other dropdown elements used in different frameworks in integrating clients (e.g. Tiki). If we can use a single dropdown class like cypht-dropdown for all dropdowns in Cypht that will be best. You can add an additional class to this specific dropdown if you need additional functionality.